app/src/main/java/com/github/se/signify/ui/screens/profile/FriendsListScreen.kt

```diff
@@ -161,21 +154,66 @@ fun FriendsListScreen(
       }
     }
 
-    // My Friends List
-    FriendListCard(
-        title = "My friends list", items = friends.value, emptyMessage = "You have no friends") {
-            friendName ->
-          FriendItem(friendName = friendName, userViewModel = userViewModel)
-        }
-    Spacer(modifier = Modifier.height(32.dp))
-
-    // New Friends Demands
-    FriendListCard(
-        title = "New friends demands",
-        items = friendsRequests.value,
-        emptyMessage = "No new friend requests") { friendRequestName ->
-          FriendRequestItem(friendRequestName = friendRequestName, userViewModel = userViewModel)
+    // Separation line
+    HorizontalDivider(
+        thickness = 1.dp,
+        color = MaterialTheme.colorScheme.onBackground.copy(alpha = 0.5f),
+        modifier = Modifier.padding(top = 24.dp, bottom = 4.dp))
+
+    // Buttons for toggling lists
+    Row(
+        modifier = Modifier.fillMaxWidth().padding(horizontal = 16.dp),
+        horizontalArrangement = Arrangement.SpaceEvenly) {
+          val numberFriends = if (friends.value.size > 1) friendsString else "Friend"
+
+          TextButton(
+              { selectedList = friendsString },
+              "${friends.value.size} $numberFriends",
+              ButtonDefaults.buttonColors(
+                  containerColor =
+                      if (selectedList == friendsString) MaterialTheme.colorScheme.primary
+                      else MaterialTheme.colorScheme.onBackground.copy(alpha = 0.5f)),
+              "${friendsString}Button")
+
+          val numberRequests =
+              if (friendsRequests.value.size > 1) friendsRequestsString else "Request"
+
+          TextButton(
```

Comment:
Instead of using this new function, you could use the one from the Utils.kt file

```suggestion
          UtilTextButton(
              onClickAction = { selectedList = friendsRequestsString },
              testTag = "${friendsRequestsString}Button",
              text = "${friendsRequests.value.size} $numberRequests",
              backgroundColor = if (selectedList == friendsRequestsString) MaterialTheme.colorScheme.primary
                  else MaterialTheme.colorScheme.onBackground.copy(alpha = 0.5f))
```